src/Microsoft.AspNetCore.Razor.Language/Intermediate/ClassDeclarationIntermediateNode.cs

```diff
@@ -16,7 +16,9 @@ public sealed class ClassDeclarationIntermediateNode : MemberDeclarationIntermed
 
         public string BaseType { get; set; }
 
-        public IList<string> Interfaces { get; set; }
+        public IList<string> Interfaces { get; set; } = new List<string>();
+
+        public IList<string> TypeParameters { get; set; } = new List<string>();
```

Comment:
Are we planning on adding a directive for this? I think that might be a start in solving @javiercn's Identity UI grossness (https://github.com/aspnet/Identity/blob/dev/src/UI/Areas/Identity/Pages/Account/Register.cshtml.cs#L49)

Reply:
Definitely not.

Supporting type parameters on type level things like Razor Pages is a whole nother can of worms and this PR does not attempt to solve this.